### ISSUE TITLE
Why does SALAD achieve higher R-Precision than ground truth but worse FID?

### ISSUE THREAD

**Issue by pidanoyes** [seokhyeonhong/salad#12]
Hi, thank you for releasing the SALAD code and for your great work on advancing text-to-motion generation. I’ve been studying the paper and experiments, and I greatly appreciate the clarity of the implementation and results.

I’m curious about the results reported for SALAD (ours). In the table, R-Precision is significantly higher than that of the ground-truth motions, while the FID score is notably worse.

Intuitively, this suggests that the generated motions follow the text even better than real human motions, yet they deviate more from the ground-truth distribution. This feels somewhat counterintuitive, as stronger text alignment would typically correlate with higher motion realism—especially given that HumanML3D is composed of real human motion data.

Could the authors clarify how this phenomenon should be interpreted?

Thanks in advance!

**Comment by seokhyeonhong** [comment]
Hello,

thank you for your interest in SALAD.

Good question! Indeed, this is one of the most common questions I've received, and I'm happy to clarify.

First of all, it's important to note that R-precision and FID measure different aspects of the generated motions, so they are not directly related. Specifically, R-precision measures how well the motion features align with the text features, and FID measures how similar the generated motions are compared to the real motions without considering input text alignment.

Meanwhile, the results on the real motions correspond to the evaluation results on the test set from HumanML3D. The evaluation models were trained on the training set of HumanML3D, and there are inherent differences between training and test set distributions. Due to this differences, the R-precision on real motions is not perfect (i.e. it's 0.8 instead of 1.0).

In this context, higher R-precision of SALAD implies that the model generates motions that are better aligned with the input text than the motions in the test set, which is a positive outcome. However, the quality of generated results does not match with the real motions in the test set, and that's why SALAD scored worse FID than the real motions.

This phenomenon is not unique to SALAD, and recent papers reported similar results: for example, [MoGenTS](https://arxiv.org/pdf/2409.17686) with FID 0.033 and R-precision-Top3 0.812. (there would have been more, but I can't remember for now)

I hope this answer helps clarification, and please let me know if you have further questions.

Best,

Seokhyeon
